sql/moz-fx-data-shared-prod/firefox_accounts_derived/fxa_users_first_seen_v1/query.sql

```diff
@@ -0,0 +1,36 @@
+WITH _current AS (
+  SELECT
+    user_id,
+    ARRAY_AGG(DISTINCT service) AS services_used,
+  FROM
+    firefox_accounts.fxa_all_events
+  WHERE
+    DATE(submission_timestamp) = @submission_date
+  GROUP BY
+    user_id
+),
+_previous AS (
+  SELECT
+    *
+  FROM
+    firefox_accounts_derived.fxa_users_first_seen_v1
```

Comment:
```suggestion
    firefox_accounts_derived.fxa_users_first_seen_v1
  WHERE
    DATE(submission_timestamp) < @submission_date
```

This is important in case we need to do a backfill of a few days. We don't want more recent (potentially incorrect) data to influence the calculation. See https://github.com/mozilla/bigquery-etl/blob/main/sql/moz-fx-data-shared-prod/telemetry_derived/clients_first_seen_v1/query.sql#L11-L21

Reply:
Hmm, this logic could break the one-row-per-client invariant. That would happen if the user is skipped on their first_seen date X, gets a row on day X+1, and we backfill just day X: they would be present for days X and X+1. My current logic would just cause the client to be present for day X+1, while what we want is the client to only be present for day X.

Reply:
Would you oppose leaving this logic out, and changing line 20 to instead be `LEAST(_previous.first_seen_date, @submission_date)`?

Reply:
> and we backfill just day X: they would be present for days X and X+1. My current logic would just cause the client to be present for day X+1, while what we want is the client to only be present for day X.

We should never do this. Since this query depends on past, any backfill operation would have to be for day X and all following days.

Reply:
Good call - forgot this is overwriting the entire table and not just the returned partitions.

Reply:
@jklukas RFAL